apps/api/src/api/application-api.ts

```diff
@@ -105,14 +106,56 @@ export const getApplicationById = async ({ applicationId }: { applicationId: num
 
 	return result;
 };
-/**
- * Gets the total of how many applications are in each state type, including a TOTAL count.
- * @param userId - The ID of the current user.
- * @returns Success with the details of the application / Failure with Error.
- */
-export const getApplicationStateTotals = async ({ userId }: { userId: string }) => {
-	const database = getDbInstance();
-	const service: ApplicationService = applicationService(database);
 
-	return await service.applicationStateTotals({ user_id: userId });
+export const approveApplication = async ({
+	applicationId,
+}: ApproveApplication): Promise<{
+	success: boolean;
+	message?: string;
+	errors?: string | Error;
+	data?: any;
+}> => {
+	try {
+		// Fetch application
+		const database = getDbInstance();
+		const service: ApplicationService = applicationService(database);
+		const result = await service.getApplicationById({ id: applicationId });
+
+		if (!result.success) return result;
+
+		const application = result.data;
+
+		const appStateManager = new ApplicationStateManager(application);
+
+		if (appStateManager.state === ApplicationStates.APPROVED) {
+			return {
+				success: false,
+				message: 'Application is already approved.',
+				errors: 'ApprovalConflict',
+			};
+		}
+
+		const approvalResult = await appStateManager.approveDacReview();
+		console.log(approvalResult);
+		if (!approvalResult.success) {
+			return {
+				success: false,
+				message: approvalResult.message || 'Failed to approve application.',
+				errors: 'StateTransitionError',
+			};
```

Comment:
`failure` is already defined so this should be: `return failure(approvalResult.message || 'Failed to approve application.', 'StateTransitionError');`